ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -2021,85 +2098,69 @@ private void ClearMerged()
-                {
-                    if (!(value is TimeSpan) && TimeSpan.TryParse(val, out tsTest))
-                        val = BaseDate.Add(tsTest).ToOADate().ToInvariantString();
-                    else
-                        val = BaseDate.Add((TimeSpan)value).ToOADate().ToInvariantString();
-
-                    _dataType = XLDataType.TimeSpan;
-                    if (style.NumberFormat.Format == String.Empty && style.NumberFormat.NumberFormatId == 0)
-                        Style.NumberFormat.NumberFormatId = 46;
-                }
-                else if (val.Trim() != "NaN" && Double.TryParse(val, XLHelper.NumberStyle, XLHelper.ParseCulture, out Double d2))
-                    _dataType = XLDataType.Number;
-                else if (DateTime.TryParse(val, out DateTime dtTest) && dtTest >= BaseDate)
-                {
-                    _dataType = XLDataType.DateTime;
+                var tuple = SetKnownTypedValue(value, style);
+                parsedValue = tuple.Item1;
+                parsed = tuple.Item2;
+            }
 
-                    if (style.NumberFormat.Format == String.Empty && style.NumberFormat.NumberFormatId == 0)
-                        Style.NumberFormat.NumberFormatId = dtTest.Date == dtTest ? 14 : 22;
-                    {
-                        DateTime forMillis;
-                        if (value is DateTime && (forMillis = (DateTime)value).Millisecond > 0)
-                        {
-                            val = forMillis.ToOADate().ToInvariantString();
-                        }
-                        else
-                        {
-                            val = dtTest.ToOADate().ToInvariantString();
-                        }
-                    }
-                }
-                else if (Boolean.TryParse(val, out Boolean bTest))
-                {
-                    _dataType = XLDataType.Boolean;
-                    val = bTest ? "1" : "0";
-                }
-                else
-                {
-                    _dataType = XLDataType.Text;
-                    if (val.Contains(Environment.NewLine) && !style.Alignment.WrapText)
-                        Style.Alignment.WrapText = true;
-                }
+            ////
+            if (!parsed)
+            {
+                // We'll have to parse it slowly :-(
+                parsedValue = DeduceCellValueByParsing(parsedValue.ToString(), style);
             }
-            if (val.Length > 32767) throw new ArgumentException("Cells can only hold 32,767 characters.");
 
-            if (SetTableHeaderValue(val)) return;
-            if (SetTableTotalsRowLabel(val)) return;
+            if (SetTableHeaderValue(parsedValue)) return;
+            if (SetTableTotalsRowLabel(parsedValue)) return;
+
+            SetInternalCellValueString(parsedValue, validate: true, parseToCachedValue: false);
+            CachedValue = null;
```

Comment:
`CachedValue` won't become null if `SetTableHeaderValue(parsedValue)` or `SetTableTotalsRowLabel(parsedValue)` succeed. I am not sure but this seems wrong. Better move it to the top where `FormulaA1` and `_richText` are reset.

Reply:
`SetTableHeaderValue` sets `XLTableField.Name`, which calls `XLCell.SetValue<T>`(T value, setTableHeader: false)`, which will set `CachedValue` to null.